Composer/packages/client/src/pages/publish/botStatusList.tsx

```diff
@@ -0,0 +1,216 @@
+import React, { useState } from 'react';
+import { Fragment } from 'react';
+import { Dropdown, IDropdownOption } from 'office-ui-fabric-react/lib/Dropdown';
+import { Spinner, SpinnerSize } from 'office-ui-fabric-react/lib/Spinner';
+
+import { navigateTo } from '../../utils/navigation';
+
+import { IStatus, PublishStatusList } from './publishStatusList';
+
+// Licensed under the MIT License.
+export interface IBotStatus {
+  id: string;
+  name: string;
+  publishTargets?: any[];
+  publishTarget?: string;
+  time?: string;
+  status?: number;
+  message?: string;
+  comment?: string;
+}
+export interface IBotStatusListProps {
+  items: IBotStatus[];
+  botPublishHistoryList: { [key: string]: any }[];
+  updatePublishHistory: (items: IStatus[], item: IBotStatus) => void;
+  updateSelectedBots: (items: IBotStatus[]) => void;
+  changePublishTarget: (PublishTarget: string, item: IBotStatus) => void;
+  onLogClick: (item: IStatus | null) => void;
+  onRollbackClick: (selectedVersion: IStatus | null, item: IBotStatus) => void;
+}
+export const BotStatusList: React.FC<IBotStatusListProps> = (props) => {
+  const {
+    items,
+    botPublishHistoryList,
+    updatePublishHistory,
+    changePublishTarget,
+    onLogClick,
+    onRollbackClick,
+  } = props;
+  const [botDescend, setBotDescend] = useState(false);
+  const sortBot = () => {
+    setBotDescend(!botDescend);
+  };
+
+  const [selectedBots, setSelectedBots] = useState<IBotStatus[]>([]);
+  const [showHistoryBots, setShowHistoryBots] = useState<string[]>([]);
+  const renderStatus = (item: IBotStatus) => {
+    if (item.status === 200) {
+      return <Icon iconName="Accept" style={{ color: 'green', fontWeight: 600 }} />;
+    } else if (item.status === 202) {
+      return (
+        <div style={{ display: 'flex' }}>
+          <Spinner size={SpinnerSize.small} />
+        </div>
+      );
+    } else {
+      return <Icon iconName="Cancel" style={{ color: 'red', fontWeight: 600 }} />;
+    }
+  };
+  const headerStyle = css({ padding: '0 32px 0 11px', fontWeight: 400, textAlign: 'left' });
+  const bodyStyle = css({ minWidth: '114px', maxWidth: '134px', padding: '11px 32px 11px 12px' });
+  const trStyle = css({ borderTop: '1px solid #EDEBE9' });
+  const spanStyle = css({ wordBreak: 'break-word' });
+
+  const RenderItem = (item: IBotStatus, index: number) => {
+    const publishTargetOptions = (): IDropdownOption[] => {
+      const options: IDropdownOption[] = [];
+      item.publishTargets &&
+        item.publishTargets.forEach((target) => {
+          options.push({
+            key: target.name,
+            text: target.name,
+          });
+        });
+      options.push({
+        key: 'manageProfiles',
+        text: formatMessage('Manage profiles'),
+        data: { style: { color: '#0078D4' } },
+      });
+      return options;
+    };
+    const onRenderOption = (option?: IDropdownOption): JSX.Element => {
+      if (!option) return <div></div>;
```

Comment:
why is it rendering empty div instead of nothing aka null?

Reply:
Refactor it.